packages/mdc-menu/foundation.js

```diff
@@ -57,593 +36,186 @@ class MDCMenuFoundation extends MDCFoundation {
-        this.adapter_.focusItemAtIndex(this.selectedIndex_);
-        return;
-      }
-
-      this.adapter_.focus();
-      // If that doesn't work, focus first item instead.
-      if (!this.adapter_.isFocused()) {
-        this.adapter_.focusItemAtIndex(0);
-      }
-    } else {
-      this.adapter_.focusItemAtIndex(focusIndex);
-    }
-  }
-
-  /**
-   * Handle clicks and cancel the menu if not a child list-item
-   * @param {!Event} evt
-   * @private
-   */
-  handleDocumentClick_(evt) {
-    let el = evt.target;
-
-    while (el && el !== document.documentElement) {
-      if (this.adapter_.getIndexForEventTarget(el) !== -1) {
-        return;
-      }
-      el = el.parentNode;
+    if (this.closeAnimationEndTimerId_) {
+      clearTimeout(this.closeAnimationEndTimerId_);
     }
 
-    this.adapter_.notifyCancel();
-    this.close(evt);
-  };
+    this.adapter_.closeSurface();
+  }
 
   /**
-   * Handle keys that we want to repeat on hold (tab and arrows).
+   * Handler function for the keydown events.
    * @param {!Event} evt
-   * @return {boolean}
-   * @private
    */
-  handleKeyboardDown_(evt) {
-    // Do nothing if Alt, Ctrl or Meta are pressed.
-    if (evt.altKey || evt.ctrlKey || evt.metaKey) {
-      return true;
-    }
+  handleKeydown(evt) {
+    const {key, keyCode} = evt;
 
-    const {keyCode, key, shiftKey} = evt;
-    const isTab = key === 'Tab' || keyCode === 9;
-    const isArrowUp = key === 'ArrowUp' || keyCode === 38;
-    const isArrowDown = key === 'ArrowDown' || keyCode === 40;
     const isSpace = key === 'Space' || keyCode === 32;
     const isEnter = key === 'Enter' || keyCode === 13;
-    // The menu needs to know if the keydown event was triggered on the menu
-    this.keyDownWithinMenu_ = isEnter || isSpace;
+    const isTab = key === 'Tab' || keyCode === 13;
 
-    const focusedItemIndex = this.adapter_.getFocusedItemIndex();
-    const lastItemIndex = this.adapter_.getNumberOfItems() - 1;
-
-    if (shiftKey && isTab && focusedItemIndex === 0) {
-      this.adapter_.focusItemAtIndex(lastItemIndex);
-      evt.preventDefault();
-      return false;
-    }
-
-    if (!shiftKey && isTab && focusedItemIndex === lastItemIndex) {
-      this.adapter_.focusItemAtIndex(0);
-      evt.preventDefault();
-      return false;
-    }
-
-    // Ensure Arrow{Up,Down} and space do not cause inadvertent scrolling
-    if (isArrowUp || isArrowDown || isSpace) {
-      evt.preventDefault();
-    }
-
-    if (isArrowUp) {
-      if (focusedItemIndex === 0 || this.adapter_.isFocused()) {
-        this.adapter_.focusItemAtIndex(lastItemIndex);
-      } else {
-        this.adapter_.focusItemAtIndex(focusedItemIndex - 1);
-      }
-    } else if (isArrowDown) {
-      if (focusedItemIndex === lastItemIndex || this.adapter_.isFocused()) {
-        this.adapter_.focusItemAtIndex(0);
-      } else {
-        this.adapter_.focusItemAtIndex(focusedItemIndex + 1);
-      }
+    if (isSpace || isEnter) {
```

Comment:
This method is really simple and readable. Most keyboard event handling functions are NOT.

Nice work!